coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CalciteTrinoOperatorTransformers.java

```diff
@@ -0,0 +1,209 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.trino.rel2trino;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+
+import com.linkedin.coral.com.google.common.base.CaseFormat;
+import com.linkedin.coral.com.google.common.base.Converter;
+import com.linkedin.coral.com.google.common.collect.ImmutableMultimap;
+import com.linkedin.coral.common.functions.Function;
+import com.linkedin.coral.common.transformers.OperatorTransformer;
+import com.linkedin.coral.common.transformers.OperatorTransformerList;
+import com.linkedin.coral.common.transformers.StandardOperatorTransformer;
+import com.linkedin.coral.hive.hive2rel.functions.HiveRLikeOperator;
+import com.linkedin.coral.hive.hive2rel.functions.StaticHiveFunctionRegistry;
+import com.linkedin.coral.trino.rel2trino.functions.TrinoElementAtFunction;
+
+
+public class CalciteTrinoOperatorTransformers extends OperatorTransformerList {
+  private static final StaticHiveFunctionRegistry HIVE_REGISTRY = new StaticHiveFunctionRegistry();
+  private static CalciteTrinoOperatorTransformers instance;
+  private CalciteTrinoOperatorTransformers() {
+  }
+
+  private CalciteTrinoOperatorTransformers(List<OperatorTransformer> operatorTransformers) {
+    super(operatorTransformers);
+  }
+
+  public static CalciteTrinoOperatorTransformers getInstance() {
+    if (instance == null) {
+      return new CalciteTrinoOperatorTransformers(initializeOperatorTransformers());
+    }
+    return instance;
+  }
+
+  private static List<OperatorTransformer> initializeOperatorTransformers() {
+    List<OperatorTransformer> operatorTransformers = new ArrayList<>();
+    // conditional functions
+    operatorTransformers.add(createOperatorTransformer(hiveToCalciteOp("nvl"), 2, "coalesce"));
+    // Array and map functions
+    operatorTransformers.add(createOperatorTransformer(SqlStdOperatorTable.ITEM, 2, TrinoElementAtFunction.INSTANCE));
```

Comment:
Did we get rid of the map and used the list instead for every entry of the map? Does it make sense to use the entire map as one entry in the list of operator transformers? FYI @ljfgem

Reply:
Yes, we got rid of the map and created transformer for each entry of the map. We can use the entire map as one entry in the list of operator transformers, but I think it's clearer to create transformer for each operator.